internal/provider/provider.go

```diff
@@ -0,0 +1,119 @@
+// Copyright © 2021 The systemk authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package provider
+
+import (
+	"context"
+	"github.com/virtual-kubelet/systemk/internal/kubernetes"
+	"io"
+	"os"
+
+	"github.com/virtual-kubelet/systemk/internal/ospkg"
+	"github.com/virtual-kubelet/systemk/internal/system"
+	"github.com/virtual-kubelet/systemk/internal/unit"
+	"github.com/virtual-kubelet/virtual-kubelet/node"
+	"github.com/virtual-kubelet/virtual-kubelet/node/api"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	stats "k8s.io/kubernetes/pkg/kubelet/apis/stats/v1alpha1"
```

Comment:
if possible lets not pull this in, a rather *not* have this functionality than pulling in k8s.io/kubernetes

Reply:
OK then we'll have to pull this one out until VK pulls in `k8s.io/kubelet v0.20.2` of newer.

Reply:
yes please

Reply:
Done!